deps/wazuh_testing/wazuh_testing/tools/sockets.py

```diff
@@ -12,7 +12,58 @@
 request_protocol = "tcp"
 
 
-def send_request(msg_request, response_size=100):
+class WazuhSocket:
+    def __init__(self, socket_file=request_socket):
+        """Encapsulate wazuh-socket communication (header with message size)
+
+        Args:
+            socket_file (str): Path of the file socket.
+        """
+        self.file = socket_file
+
+    def send(self, msg, response_size=4):
+        """Send and receive data to wazuh-socket (header with message size)
+
+        Args:
+            msg (str): data to send
+
+        Returns:
+            str: received data
+        """
+        try:
+            wazuh_socket = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            encoded_msg = msg.encode('utf-8')
+            request_msg = struct.pack("<I", len(encoded_msg)) + encoded_msg
+
+            @retry(socket.error)
+            def connection(_socket, request):
+                _socket.connect(request)
+
+            @retry(socket.error)
+            def send_msg(_socket, msg):
+                _socket.send(msg)
+
+            @retry(ValueError)
+            def recv_response(_socket, size):
+                size = struct.unpack("<I", _socket.recv(size, socket.MSG_WAITALL))[0]
+                recv_msg = _socket.recv(size, socket.MSG_WAITALL)
+                if recv_msg == '':
+                    raise ValueError
+                return recv_msg
```

Comment:
I think these definitions could at least moved output of the `try` block.

Reply:
Done 29484935443c9d15880a267f5abcd72f8779aa3a